app/src/sharedTest/java/org/oppia/android/app/profile/ProfileChooserFragmentTest.kt

```diff
@@ -207,6 +212,7 @@ class ProfileChooserFragmentTest {
   fun testProfileChooserFragment_initializeProfiles_checkProfilesLastVisitedTimeIsShown() {
     profileTestHelper.initializeProfiles()
     launch<ProfileChooserActivity>(createProfileChooserActivityIntent()).use {
+      testCoroutineDispatchers.runCurrent()
```

Comment:
@BenHenning  There are 4 Ignore test cases like this one, which performs the opening of next activity which actually not supported by Robolectric. Am I missing something here or should we this test espresso-only?

This test case is something which cannot be broken down into pieces to test things separately as this is a collective step to test which require hopping between different screens.

Reply:
Workaround -  https://github.com/libon/RobolectricMultipleActivityTestIssue/pull/1/files

Reply:
How do the test coroutine dispatchers impact this? As I understand it, this is a Robolectric limitation since Robolectric isn't designed to switch between activities.

I'm also not sure I buy that this can't be broken up. This particular test can be split into several:
- testProfileChooserFragment_initialProfiles_clickFirst_opensPinPasswordActivity
- testPinPasswordActivity_inputCorrectPin_opensHomeActivity
- testHomeActivity_validatedProfile_pressBack_closesActivity
- testProfileChooserFragment_afterVisitingHomeActivity_showsJustNowText

Each of these are verifying a single activity, and verify boundaries by:
1. Using an intent matcher to check if a particular activity _would_ be open
2. Ensuring the state for the test is set up equivalently to if you did visit a previous activity (such as in the case of the last test above)

Note that this is generally better testing: it keeps things isolated, and it's verifying a single behavior at one time. Not all of the tests above belong in this suite either--they should be in each corresponding activity's test suite.

You could make the argument that this isn't as thorough since we aren't testing integration boundaries--that's true. We get very *close* to correct with this approach, but this does not replace proper integration/e2e tests. That's fine; we'll be implementing e2e tests in the future. Espresso should be used for unit tests based on our current team practices.

Does this help?

Reply:
yes, divided the test cases. Done.